### PR TITLE
Expanded CI builds

### PR DESCRIPTION
**Pull request by mdiep**
- Log the Swift version
- Test Swift 4 builds

### CODE REVIEW

**Comment by andersio** [comment]
https://github.com/apple/swift-evolution/blob/master/proposals/0151-package-manager-swift-language-compatibility-version.md

Looks like this is the way for a 3.2 build.

**Comment by mdiep** [comment]
Yup. I just learned that in a WWDC lab. ☺️

File: .travis.yml [hunk 1]
```diff
@@ -49,6 +49,7 @@ jobs:
       env:
         - JOB=PODSPEC
     - script:
+      - swift --version
```
Comment (sharplet) [inline .travis.yml]: indentation looks off here

Comment (mdiep) [inline .travis.yml]: Thanks! 😅

**Comment by mdiep** [comment]
The tests need some more updates for Swift 4.

**Comment by mdiep** [comment]
Ready!

**Comment by andersio** [comment]
Instead of changing `apply()` to `apply(())`, we should fix it by adding an overload for `Input == Void`.

**Pull request merged**